Document Consul ECS Resource Usage (#12228)
hashicorp/consul: website/content/docs/ecs/resource-usage.mdx

---
layout: docs
page_title: Resource Usage - AWS ECS
description: >-
  Resource usage of the Consul Service Mesh on AWS ECS (Elastic Container Service).
---

# Resource Usage

> We ran performance tests on Consul on ECS to determine the resource usage of

Reviewer: ```suggestion
We ran performance tests of Consul on ECS to determine the resource usage of
```

the sidecar containers Consul on ECS injects along with the ACL controller.
[The architecture page](/docs/ecs/architecture) describes each Consul on ECS
component in depth.

> 

Reviewer 2: Would the paragraph starting on line15 be easier to consume in list format, i.e.:

We used the following procedure to measure resource usage:
* Executed performance tests while deploying clusers of various sizes. We ensured that deployment conditions stressed Consul on ESC components.
* After each performance test session, we recorded resource usage for each component to determine worst-case scenario resource usage in a production environment. 
* We used Fargate's minimum allowed CPU (256 shares) and memory settings (512 MB) on ECS during performance testing to demonstrate that Consul on ECS along with application containers can run on the smallest ECS tasks.

We ran performance tests by deploying clusters of varying sizes and measuring
resource usage during conditions that stress Consul on ECS components. After
each iteration of performance testing, we recorded resource usages of each
component to determine worst-case scenario resource usage in a production

> environment. During performance testing, we used ECS' minimum allowable CPU and memory
> resources, 256 CPU shares and 512 MB of memory, to demonstrate that

Reviewer: ```suggestion
environment. During performance testing, we used Fargate's minimum allowed CPU and memory
settings on ECS, 256 CPU shares and 512 MB of memory, to demonstrate that
```

nitpicking: the EC2 launch type doesn't restrict you to the Fargate cpu/memory configurations

Consul on ECS along with application containers can run on the smallest ECS

> tasks. Here is the maximum resource usage we observed for each container:

Reviewer: ```suggestion
tasks. The following table shows the maximum resource usage we observed for each container:
```

One of my doc learnings from Adam is to make the document read sort of linearly downward. So more explaining what follows, and less "as shown above" where the reader has to jump back up.

| Container      | CPU | Memory |
| -------------- | --- | ------ |
| ACL Controller | 5%  | 43 MB  |
| Health Sync    | 6%  | 35 MB  |
| Consul Client  | 10% | 87 MB  |
| Envoy          | 18% | 55 MB  |

> As shown above, Consul on ECS’ containers use well under the minimum CPU and

Reviewer: ```suggestion
The containers added by Consul on ECS use well under the minimum CPU and
```

Reviewer 2: ```suggestion
The containers added by Consul on ECS consume resources well below the minimum CPU and
```
I'm not sure if that changes the meaning - in any case, "below" is probably the correct diction.

> memory limits for an ECS Task. `memory` and `cpu` can be passed into Mesh Init
> if additional resources are necessary.

Reviewer: ```suggestion
memory limits for an ECS Task. Use the `memory` and `cpu` settings for the task definition
if additional resources are necessary for your application task.
```

Clarifying slightly that they should only need additional cpu/memory for the Consul ECS containers based on their application's needs.
